core/frontend/src/UserPreferences.ts

```diff
@@ -9,76 +9,56 @@
 
 import { AccessToken, GuidString } from "@itwin/core-bentley";
 
-/** Argument for methods that can supply an iTwinId and iModelId.
- * @beta
+/** Arguments supplied to [[UserPreferencesAccess]] methods to specify a single user preference.
+ * @public
  */
-export interface ITwinIdArg {
-  readonly iTwinId?: GuidString;
-  readonly iModelId?: GuidString;
-}
-
-/** Argument for methods that can supply an access token.
- * @beta
- */
-export interface TokenArg {
+export interface UserPreferenceKeyArgs {
+  /** If supplied, indicates the user preference is defined at the level of this iTwin. */
+  iTwinId?: GuidString;
+  /** If supplied, indicates the user preference is defined at the level of this iModel. */
+  iModelId?: GuidString;
   accessToken?: AccessToken;
+  /** A unique namespace for [[key]]. */
+  namespace?: string;
+  /** The key that identifies the user preference. */
+  key: string;
 }
 
-/** Argument for methods that can supply the user preference content.
- * @beta
+/** Arguments supplied to [[UserPreferencesAccess.save]].
+ * @public
  */
-export interface PreferenceArg extends PreferenceKeyArg {
-  readonly content?: any;
-}
-
-/** Argument for methods that must supply a key for the user preference.
- * @beta
- */
-export interface PreferenceKeyArg {
-  readonly namespace?: string;
-  readonly key: string;
+export interface SaveUserPreferenceArgs extends UserPreferenceKeyArgs {
+  /** The value to associate with the user preference. */
```

Comment:
I'm guessing that the value needs to be JSON-encodable. It may be good to mention that fact, although maybe that is just assumed by users.

Reply:
All I currently have to go on are the docs @kabentley wrote. Somebody somewhere has implemented this interface. @aruniverse @grigasp do you know who?

I have other questions like, what happens if you specify neither an iTwinId nor an iModelId in UserPreferenceKeyArgs? Presumably nothing, unless there's some global user preferences level above iTwin.

Reply:
`UserPreferencesAccess` is used by the [maplayers widget](https://github.com/iTwin/viewer-components-react/blob/caa244bee5585fb6ac6ddd6cdb89e8445c2b7f5f/packages/itwin/map-layers/src/mapLayers.ts#L12) to persist MapLayerSource objects.  Initially it was stored directly into PSS, but was generalized with this interface. 
@calebmshafer May have further details.

Reply:
Pineapple [implements](https://bentleycs.visualstudio.com/beconnect/_git/TCDEAppService?path=packages%2Fmodules%2Fpineapple-preferences-access-client%2Fsrc%2FPreferencesAccessClient.ts) it.

Reply:
I reverted the changes to user preferences. These APIs need polishing up. I will file a separate issue.